### ISSUE TITLE
Auto-sklearn Classifier: Dealing with class imbalance

### ISSUE THREAD

**Issue by deepikaynm** [automl/auto-sklearn#470]
Hello,

I am trying to use auto-sklearn with a heavily imbalanced dataset. 
Is it possible to incorporate over- and under-sampling techniques with cross-validation resampling? 

I noticed the same issue was closed before: [https://github.com/automl/auto-sklearn/issues/113](113)
Since SMOTE is now available through imblearn API, I was wondering if this is already implemented? 

As an example, this is possible in the caret package in R.

ctrl <- trainControl(method = "repeatedcv", 
                     number = 10, 
                     repeats = 10, 
                     verboseIter = FALSE,
                     sampling = "smote")

set.seed(42)
model_rf_smote <- caret::train(classes ~ .,
                              data = train_data,
                              method = "rf",
                              preProcess = c("scale", "center"),
                              trControl = ctrl)

Thanks in advance,
Deepika

**Comment by mfeurer** [comment]
The original issue in scikit-learn is still open (i.e. the pipeline object cannot handle over and undersampling methods). As we're only wrapping scikit-learn (and not its extensions) this is not possible.
